chain/types/actor_event.go

```diff
@@ -0,0 +1,72 @@
+package types
+
+import (
+	"github.com/ipfs/go-cid"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-state-types/abi"
+)
+
+type ActorEventBlock struct {
+	// The value codec to match when filtering event values.
+	Codec uint64 `json:"codec"`
+
+	// The value to want to match on associated with the corresponding "event key"
+	// when filtering events.
+	// Should be a byte array encoded with the specified codec.
+	// Assumes base64 encoding when converting to/from JSON strings.
+	Value []byte `json:"value"`
+}
+
+type SubActorEventFilter struct {
+	Filter ActorEventFilter `json:"filter"`
+
+	// If true, all available matching historical events will be written to the response stream
+	// before any new real-time events that match the given filter are written.
+	// If `Prefill` is true and `FromEpoch` is set to latest, the pre-fill operation will become a no-op.
+	// if `Prefill` is false and `FromEpoch` is set to earliest, historical events will still be sent to the client.
+	Prefill bool `json:"prefill"`
+}
+
+type ActorEventFilter struct {
+	// Matches events from one of these actors, or any actor if empty.
+	// For now, this MUST be a Filecoin address.
+	Addresses []address.Address `json:"addresses,omitempty"`
+
+	// Matches events with the specified key/values, or all events if empty.
+	// If the value is an empty slice, the filter will match on the key only, accepting any value.
+	Fields map[string][]ActorEventBlock `json:"fields,omitempty"`
+
+	// Interpreted as an epoch (in hex) or one of "latest" for last mined block, "earliest" for first,
+	// Optional, default: "latest".
+	FromEpoch string `json:"fromEpoch,omitempty"`
+
+	// Interpreted as an epoch (in hex) or one of "latest" for last mined block, "earliest" for first,
+	// Optional, default: "latest".
+	ToEpoch string `json:"toEpoch,omitempty"`
```

Comment:
Epoch -> Height for consistency.

Reply:
Need to decide what to do if this and/or `From` will cause an error if they are later than the current heaviest. If we're going to do that with earliest eventually then it may be something to consider. It just means you can't be sloppy with your calls other than `nil`.

Reply:
> If we're going to do that with earliest eventually then it may be something to consider.

I'd say yes? You can just specify "nil" if you don't care.